### PR TITLE
Settings UI: Put advanced settings in foldable cards

### LINKED ISSUE #6503: Move all of stats settings into a foldable card
#### Current
image

#### Proposed

Foldable card:

Card title: Site stats
Card description: TBD

All of the settings should be in the current order, but in the expanded view only.

### LINKED ISSUE #6507: Move all spelling and grammar settings into a foldable card
#### Current
image

#### Proposal
This will be a similar solution to https://github.com/Automattic/jetpack/issues/6504

Card title: none
Primary setting: Check your spelling, style, and grammar

The rest of the settings should remain as is, but in the expanded view instead of advanced options.

### PR DESCRIPTION
**Pull request by dereksmart**
Fixes #6507
Fixes #6503

❗️ Make sure to `yarn distclean` && `yarn build` to get the newest dops-components❗️ 

screen shot 2017-03-02 at 10 20 35 am

screen shot 2017-03-02 at 10 20 45 am

To Test: 
Make sure forms and saving works properly.  
Make sure it looks good.

### CODE REVIEW

**Comment by eliorivero** [comment]
When you click the chevron, the page jumps to a blank page.
`wp-admin/admin.php?page=jetpack#/writing` jumps to `wp-admin/admin.php?#/writing`
`wp-admin/admin.php?page=jetpack#/traffic` jumps to `wp-admin/admin.php?#/traffic`

**Comment by dereksmart** [comment]
Turns out the actual designs we're going for is this 
screen shot 2017-03-02 at 9 09 18 am

In progress for now.

**Comment by eliorivero** [comment]
Makes sense. The advanced settings looked integrated in the Stats card, as part of the card, but they are a bit confusing in the Composing card since it's not clear if they're advanced options for all the card or just one setting.
I wonder how it looks in Stats. Because in Composing now it's clear it's for AtD, but Stats has two options and adding the chevron for one of them might look like it's only for that option and would be confusing.

**Comment by dereksmart** [comment]
> I wonder how it looks in Stats. Because in Composing now it's clear it's for AtD, but Stats has two options and adding the chevron for one of them might look like it's only for that option and would be confusing.

I was thinking this too.  @MichaelArestad?

**Comment by MichaelArestad** [comment]
@dereksmart for stats, we will put all of the settings in the foldable part. There will just be a normal description and the chevron for stats. Does that make sense?

**Comment by dereksmart** [comment]
Will there be a toggle for activating the module? Can you please just whip up a quick screenshot/mock?

**Comment by MichaelArestad** [comment]
@dereksmart no toggle. There already is no toggle for this module.

Something like this (but the JP version)

image

**Comment by eliorivero** [comment]
This PR is now updated and AtD and Stats are now in foldable cards:

captura de pantalla 2017-03-09 a las 17 59 38
captura de pantalla 2017-03-09 a las 17 59 52
captura de pantalla 2017-03-09 a las 18 11 14
captura de pantalla 2017-03-09 a las 18 11 01

File: _inc/client/traffic/site-stats.jsx [hunk 1]
```diff
@@ -85,11 +85,10 @@ export const SiteStats = moduleSettingsForm(
 				unavailableInDevMode = this.props.isUnavailableInDevMode( 'stats' ),
 				siteRoles = this.props.getSiteRoles();
 			return (
-				<SettingsCard
-					{ ...this.props }
-					hideButton
+				<FoldableCard
 					header={ __( 'Site stats' ) }
-					module="stats">
+					className={ classNames( 'jp-foldable-settings-standalone', { 'jp-foldable-settings-disable': unavailableInDevMode } ) }
```
Comment (dereksmart) [inline _inc/client/traffic/site-stats.jsx]: I think this one makes sense to have a clickableHeader

**Comment by dereksmart** [comment]
Pushed one small change to make stats card header clickable.  I think this is good.  @MichaelArestad what say you?

**Comment by MichaelArestad** [comment]
This is great! Thank you! Ship this big ol' thang!

File: _inc/client/components/settings-card/style.scss [hunk 2]
```diff
@@ -3,46 +3,67 @@
...
+			padding-left: rem( 24px );
+			padding-right: rem( 24px );
+		}
 
-	.dops-foldable-card__header-text {
-		font-size: rem( 14px );
-	}
+		.form-toggle__switch {
+			float: left;
+			margin-top: 2px;
```
Comment (MichaelArestad) [inline _inc/client/components/settings-card/style.scss]: Needs rem()

File: _inc/client/components/settings-card/style.scss [hunk 3]
```diff
@@ -3,46 +3,67 @@
...
+				max-width: 85%;
+			}
+		}
+
+		.dops-foldable-card__action {
+			right: rem( 10px );
+		}
+
+		.jp-module-settings__learn-more {
+			right: 2px;
```
Comment (MichaelArestad) [inline _inc/client/components/settings-card/style.scss]: Needs rem()

File: _inc/client/components/settings-group/style.scss [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	position: relative;
+	margin-bottom: 0;
+
+	p {
+		font-size: rem( 14px );
+		margin-top: 0;
+		margin-bottom: rem( 24px );
+	}
+	.form-toggle__label {
+		margin-top: 4px;
```
Comment (MichaelArestad) [inline _inc/client/components/settings-group/style.scss]: both of these need rem()

**Pull request merged**